src/controllers/satisfactioncontroller.js

```diff
@@ -31,16 +30,19 @@ export default class SatisfactionController extends Controller
      */
     updateCustomerSatisfaction()
     {
-        GAME.model.customers.forEach(function (customer) {
+        GAME.model.customers.forEach(function(customer) {
             customer.satisfaction -= 10;
 
-            const customerController = new CustomerController();
-
             let config = GAME.model.config;
 
-            if (customer.satisfaction <= config.sendAwayThreshold)
+            if (customer.satisfaction <= config.sendAwayThreshold) {
+                const customerController = new CustomerController();
+                const satisfactionController = new SatisfactionController();

```

Comment:
I made a new SatisfactionController, even though this is inside the satisfaction controller. So Maybe this should be done otherwise? Didn't know how though.

Reply:
Use `this`.

`this.updatePlayerSatisfaction(customer)`.

Reply:
Yeah I tried that but then it gets buggy and gives the error:

Uncaught TypeError: Cannot read property 'updatePlayerSatisfaction' of undefined

Reply:
It's not buggy. This happens because you still use `function` syntax, which declares a function-level `this` argument.

Moving to ES6 arrow syntax should resolve this.

Reply:
Ahhh alright :D